specification.md

```diff
@@ -157,406 +157,333 @@ The Serverless Workflow language is composed of:
+      "output": "${ .provisionedOrders }"
+    }
 }
 ```
 
-Note that the referenced function definition type in this case must be `rest` (default type). 
+</td>
+<td valign="top">
 
-For more information about functions, reference the [Functions definitions](#Function-Definition) section.
+```yaml
+stateDataFilter:
+  input: "${ .orders }"
+  output: "${ .provisionedOrders }"
+```
 
-### Using Functions For RPC Service Invocations
+</td>
+</tr>
+</table>
 
-Similar to defining invocations of operations on RESTful services, you can also use the workflow 
-[functions definitions](#Function-Definition) that follow the remote procedure call (RPC) protocol.
-For RPC invocations, the Serverless Workflow specification mandates that they are described using [gRPC](https://grpc.io/),
-a widely used RPC system. 
-gRPC uses [Protocol Buffers](https://developers.google.com/protocol-buffers/docs/overview) to define messages, services,
-and the methods on those services that can be invoked. 
+</details>
 
-Let's look at an example of invoking a service method using RPC. For this example let's say we have the following
-gRP prototocol buffer definition in a myuserservice.proto file:
+State data filters can be used to filter the states data input and output.
 
-```text
-service UserService {
-    rpc AddUser(User) returns (google.protobuf.Empty) {
-        option (google.api.http) = {
-            post: "/api/v1/users"
-            body: "*"
-        };
-    }
-    rpc ListUsers(ListUsersRequest) returns (stream User) {
-        option (google.api.http) = {
-            get: "/api/v1/users"
-        };
-    }
-    rpc ListUsersByRole(UserRole) returns (stream User) {
-        option (google.api.http) = {
-            get: "/api/v1/users/role"
-        };
-    }
-    rpc UpdateUser(UpdateUserRequest) returns (User) {
-        option (google.api.http) = {
-            patch: "/api/v1/users/{user.id}"
-            body: "*"
-        };
+The state data filters `input` property expression is applied when the workflow transitions to the current state and receives its data input.
+It can be used to select only data that is needed and disregard what is not needed.
+If `input` is not defined or does not select any parts of the state's data input, its data input is not filtered.
+
+The state data filter `output` property expression is applied right before the state transitions to the next state defined. 
+It filters the state's data output to be passed as data input to the transitioning state.
+If the current state is the workflow end state, the filtered state's data output becomes the workflow data output.
+If `output` is not defined or does not select any parts of the state's data output, its data output is not filtered.
+
+Results of the `input` expression should become the state data. 
+Results of the `output` expression should become the state data output.
+
+For more information on this you can reference the [data merging](#Data-Merging) section.
```

Comment:
Nit; I think it can be pretty confusing that this mentions merging when no merging is done for state data filters...